atuin/src/command/client/search.rs

```diff
@@ -123,7 +124,13 @@ impl Cmd {
 
         if self.interactive {
             let item = interactive::history(&self.query, settings, db).await?;
-            eprintln!("{item}");
+
+            if settings.exit_mode == ExitMode::Execute {
+                let mut cmd = shell(item);
```

Comment:
I think we're going to need to ensure that Atuin executes the command as a child of the shell, and not of Atuin. We'll need to find some sort of structured data that can be returned to the shell plugin, parsed there, and then trigger execution or just insert with the line editor

That should also solve the aliases issue

Reply:
Oh, good point!
That would also explain, what I just noticed yesterday: those direct executions were not logged in the history.

I also started looking for such control character, but haven't found anything, yet.
I tried writing `\n` as byte directly to stdout, but that didn't work.

If I find something, it would probably be best to move this to its own `fn` so it's easier to `#[cfg(unix)]`/`#[cfg(windows)]`.

Reply:
You could make it exit w/ different exit codes catch that in the plugin and if exit code == x run the command

Reply:
As a temporary solution I implemented the fallback if the exit code is non-zero.
Little caveat atm is, that for some reason it looses colorinfo through the pipe.

And eventually I would prefer if we could run it as an actual child of the parent shell to the current atuin instance, but I need more time to research on this..